### PR TITLE
Add check for /etc/fstab

### PR DESCRIPTION
**Pull request by soapy1**
fixes https://github.com/hashicorp/vagrant/issues/11894

### CODE REVIEW

**Review by dcermak — APPROVED** [review]
I have tested this locally with a box that lacked a /etc/fstab and the issue is resolved by that.

File: plugins/guests/linux/cap/persist_mount_shared_folder.rb [hunk 1]
```diff
@@ -53,6 +61,10 @@ def self.persist_mount_shared_folder(machine, folders)
 
         private
 
+        def self.fstab_exists?(machine)
+          machine.communicate.test("stat /etc/fstab")
```
Comment (chrisroberts) [inline plugins/guests/linux/cap/persist_mount_shared_folder.rb:65]: Can we simplify this down to just `test -f /etc/fstab` ?

File: plugins/guests/linux/cap/persist_mount_shared_folder.rb [hunk 2]
```diff
@@ -8,14 +8,22 @@ module Cap
...
+        @@logger = Log4r::Logger.new("vagrant::guest::linux::persist_mount_shared_folders")
+
         # Inserts fstab entry for a set of synced folders. Will fully replace
         # the currently managed group of Vagrant managed entries. Note, passing
         # empty list of folders will just remove entries
         #
         # @param [Machine] machine The machine to run the action on
         # @param [Map<String, Map>] A map of folders to add to fstab
         def self.persist_mount_shared_folder(machine, folders)
+          unless fstab_exists?(machine)
```
Comment (chrisroberts) [inline plugins/guests/linux/cap/persist_mount_shared_folder.rb:20]: Update the `unless` to an `if !fstab_exists?` simply due to the overall preference for `if` based conditionals

**Review by chrisroberts — CHANGES_REQUESTED** [review]

**Review by chrisroberts — APPROVED** [review]

**Comment by marga-kinvolk** [comment]
Thanks for working on this fix!

I do find the logic of this change a bit strange. If `/etc/fstab` doesn't exist, it bails out of doing the additions that it was planning to do. It would be fine to create an `/etc/fstab` that doesn't exist. At least in Flatcar it doesn't exist because all the OS mounts are managed by systemd, not because creating it doesn't work.

In other words, I think the check should be inside `remove_vagrant_managed_fstab` rather than on the whole function.

Thanks again.

**Comment by soapy1** [comment]
@marga-kinvolk thanks for following up about this!
It seems I have misunderstood the interactions between `/etc/fstab` and systemd. It seems desirable to allow Vagrant to modify fstab regardless of if systemd is being used to manage mounts. Going to push up a change in a little bit 🚀 

ref: https://docs.flatcar-linux.org/os/kernel-modules/#create-a-writable-overlay

**Pull request merged**